test/common_test.go

```diff
@@ -319,6 +319,18 @@ func (c *T) RecvStart() bool {
 		return false
 	}
 	defer m.Free()
+	if addr := m.Port.Address(); addr != c.addr {
+		c.Debugf("Got unexpected message port address: %s", addr)
+		return false
+	}
+	if !m.Port.IsOpen() {
+		c.Debugf("Expected message port open")
+		return false
+	}
+	if m.Port.IsServer() || !m.Port.IsClient() {
```

Comment:
In fact, I'm pretty sure this is wrong.  This gets called in some of the test cases for *both* sides of the connection (e.g. Pair, which I note is one of the failing test cases.)